### PR TITLE
Move disk related functions to pkg/disk

### PR DESCRIPTION
**Pull request by faisal-memon**
Refactor that consolidates all disk function in the pkg/ folder.

### CODE REVIEW

**Review by kfox1111 — COMMENTED** [review]
Looks good to me.

File: pkg/disk/json.go [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+)
+
+// WriteJWTBundleSet write the given JWT bundles to disk
+func WriteJWTBundleSet(jwkSet *jwtbundle.Set, dir string, jwtBundleFilename string) error {
+	var errs []error
+	bundles := make(map[string]interface{})
+	for _, bundle := range jwkSet.Bundles() {
+		bytes, err := bundle.Marshal()
+		if err != nil {
+			errs = append(errs, fmt.Errorf("Unable to marshal JWT bundle: %w", err))
```
Comment (MarcosDY) [inline pkg/disk/json.go:22]:
```suggestion
			errs = append(errs, fmt.Errorf("unable to marshal JWT bundle: %w", err))
```

File: pkg/disk/json.go [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+			continue
+		}
+		bundles[bundle.TrustDomain().Name()] = base64.StdEncoding.EncodeToString(bytes)
+	}
+
+	if err := writeJSON(bundles, dir, jwtBundleFilename); err != nil {
+		errs = append(errs, fmt.Errorf("Unable to write JSON file: %w", err))
```
Comment (MarcosDY) [inline pkg/disk/json.go:29]:
```suggestion
		errs = append(errs, fmt.Errorf("unable to write JSON file: %w", err))
```

File: pkg/disk/json.go [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+}
+
+// WriteJWTBundle write the given JWT SVID to disk
+func WriteJWTSVID(jwtSVID *jwtsvid.SVID, dir, jwtSVIDFilename string) error {
+	filePath := path.Join(dir, jwtSVIDFilename)
+
+	return os.WriteFile(filePath, []byte(jwtSVID.Marshal()), os.ModePerm)
+}
+
+func writeJSON(certs map[string]interface{}, dir, filename string) error {
```
Comment (MarcosDY) [inline pkg/disk/json.go:42]:
```suggestion
func writeJSON(certs map[string]any, dir, filename string) error {
```

File: pkg/disk/json_test.go [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+	jwtSVIDFilename   = "jwt.json"
+)
+
+func TestWriteJWTBundleSet(t *testing.T) {
+	td, err := spiffeid.TrustDomainFromString("example.test")
+	require.NoError(t, err)
+
+	jwtBundle := jwtbundle.New(td)
+	jwtBundleSet := jwtbundle.NewSet(jwtBundle)
+	require.True(t, jwtBundleSet.Has(td))
```
Comment (MarcosDY) [inline pkg/disk/json_test.go:35]: NIT: sound to me like this validation is out of scope, it is validating a go-spiffe validation, anyway, you can keep it if you prefer :D

File: pkg/disk/json_test.go [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+	err = WriteJWTBundleSet(jwtBundleSet, tempDir, jwtBundleFilename)
+	require.NoError(t, err)
+
+	actualJWTBundle, err := jwtbundle.Load(td, path.Join(tempDir, jwtBundleFilename))
+	require.NoError(t, err)
+	require.Equal(t, jwtBundle, actualJWTBundle)
+}
+
+func TestWriteJWTSVID(t *testing.T) {
+	spiffeID, err := spiffeid.FromString("spiffe://example.test/workload")
```
Comment (MarcosDY) [inline pkg/disk/json_test.go:48]: NIT: here you can use  spiffeid.RequireFromString

File: pkg/disk/x509.go [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+// the svid, key and bundle of certificates.
+// It is possible to change output setting `addIntermediatesToBundle` as true.
+func WriteX509Context(x509Context *workloadapi.X509Context, addIntermediatesToBundle, includeFederatedDomains bool, certDir, svidFilename, svidKeyFilename, svidBundleFilename string) error {
+	svidFile := path.Join(certDir, svidFilename)
+	svidKeyFile := path.Join(certDir, svidKeyFilename)
+	svidBundleFile := path.Join(certDir, svidBundleFilename)
+
+	// There may be more than one certificate, but we're only interested in the default one
+	svid := x509Context.DefaultSVID()
+	certs := svid.Certificates
```
Comment (MarcosDY) [inline pkg/disk/x509.go:29]: ca nyou move certs definition near to addIntermediatesToBundle, first time it is used, to simplify code view?

File: pkg/disk/x509.go [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+
+	// Extract bundle for the default SVID
+	bundleSet, found := x509Context.Bundles.Get(svid.ID.TrustDomain())
+	if !found {
+		return fmt.Errorf("no bundles found for %s trust domain", svid.ID.TrustDomain().String())
```
Comment (MarcosDY) [inline pkg/disk/x509.go:34]:
```suggestion
		return fmt.Errorf("no bundles found for %q trust domain", svid.ID.TrustDomain().String())
```

File: pkg/disk/x509_test.go [hunk 8]
```diff
@@ -0,0 +1,142 @@
...
+
+			bundle := test.ca.Roots()
+
+			if test.intermediateInBundle {
+				bundle = append(bundle, certs[1:]...)
+				certs = certs[:1]
+			}
+
+			if test.includeFederatedDomains {
+				federatedSpiffeID, err := spiffeid.FromString("spiffe://federated.test/server")
```
Comment (MarcosDY) [inline pkg/disk/x509_test.go:117]: why not creating a spiffeid.TrustDomain instead? you dont need the spiffeid here

File: pkg/disk/x509_test.go [hunk 9]
```diff
@@ -0,0 +1,142 @@
...
+				require.NoError(t, err)
+				x509Context.Bundles.Add(x509bundle.FromX509Authorities(federatedSpiffeID.TrustDomain(), test.federatedCA.Roots()))
```
Comment (MarcosDY) [inline pkg/disk/x509_test.go:119]: I think bundle MUST always include the federated bunde, because if now how can we validate that federated bundle is not included in file because we didnt get it and no because includeFederatedDomains is not set?

File: pkg/disk/json_test.go (hunk 5 above)
Comment (faisal-memon) [reply pkg/disk/json_test.go]: done

File: pkg/disk/json_test.go (hunk 4 above)
Comment (faisal-memon) [reply pkg/disk/json_test.go]: Removed

File: pkg/disk/x509.go (hunk 6 above)
Comment (faisal-memon) [reply pkg/disk/x509.go]: done

File: pkg/disk/x509_test.go (hunk 8 above)
Comment (faisal-memon) [reply pkg/disk/x509_test.go]: good catch, done

File: pkg/disk/x509_test.go (hunk 9 above)
Comment (faisal-memon) [reply pkg/disk/x509_test.go]: Good catch. Added new unit test to cover this case.

File: pkg/disk/json.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (faisal-memon) [reply pkg/disk/json.go]: Fixed

**Review by MarcosDY — APPROVED** [review]
looks good, but can you resolve DCO?
